grouper/audit.py

```diff
@@ -73,6 +73,7 @@ def assert_controllers_are_auditors(group: Group) -> None:
         cur_group = queue.pop()
         if cur_group in checked:
             continue
+        checked.add(cur_group)
         details = graph.get_group_details(cur_group)
         for chk_user, info in details["users"].items():
             if chk_user in checked:
```

Comment:
nit: we could also add this check before we append to the queue to avoid processing groups multiple times, but I guess efficiency isn't super critical here